### PR TITLE
Update certificates (October 2024)

### PR DESCRIPTION
**Pull request by iharthi**
* Add autoupdater script 
* Update to use the newest certificates
* Add all valid certificates from TSA certificates tab on SK's website
* Remove sk-ocsp-responder-certificates from the package, as they expired (and all OCSP certificates are listed as expired)

### CODE REVIEW

File: autoupdate/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+## Usage
+
+```shell
+cd autoupdate
+just update
+```
+
+## Steps to take manually after autoupdate
+* Check that the certificates are correct. Never you should commit and publish a package without checking that
```
Comment (JohanViirok) [inline autoupdate/README.md]:
```suggestion
* Check that the certificates are correct. Never should you commit and publish a package without checking that
```

File: autoupdate/update.py [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+        self.dry_run = dry_run
+        self._certificates: list[Certificate] = []
+
+    @cached_property
+    def document(self):
+        return bs4.BeautifulSoup(
+            requests.get(
+                self.url,
+                headers={
+                    "User-Agent": "python-esteid-certificates updater; https://github.com/thorgate/esteid-certificates",
```
Comment (JohanViirok) [inline autoupdate/update.py:56]: praise: That user agent is a nice touch 🤌

File: autoupdate/update.py [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+                n_root_certs := len(
+                    [
+                        cert
+                        for cert in self._certificates
+                        if cert.certificate_type == CertificateType.ROOT
+                    ]
+                )
+            )
+            == 1
+        ):
```
Comment (JohanViirok) [inline autoupdate/update.py]: question: So nested and complex. Wouldn't it be much simpler like this?
```python
n_root_certs = len([cert for cert in self._certificates if cert.certificate_type == CertificateType.ROOT])
if n_root_certs != 1:
```

File: esteid_certificates/constants.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    "TEST of SK ID Solutions EID-Q 2024E": "TEST_of_SK_ID_Solutions_EID-Q_2024E.pem",
+    "TEST of SK ID Solutions EID-Q 2024R": "TEST_of_SK_ID_Solutions_EID-Q_2024R.pem",
+    "TEST of SK ID Solutions ORG 2021E": "TEST_ORG_2021E.pem",
+    "TEST of SK ID Solutions ORG 2021R": "TEST_ORG_2021R.pem",
+    "DEMO SK TIMESTAMPING AUTHORITY 2020": "DEMO_SK_TIMESTAMPING_AUTHORITY_2020.pem",
+    "TEST of SK TSA CA 2023E": "TEST_of_SK_TSA_CA_2023E.pem",
+    "TEST of SK TSA CA 2023R": "TEST_of_SK_TSA_CA_2023R.pem",
+    "DEMO SK TIMESTAMPING AUTHORITY 2023E": "demo_tsu_ecc_2023.pem",
+    "DEMO SK TIMESTAMPING AUTHORITY 2023R": "demo_tsu_rsa_2023.pem",
+}
```
Comment (JohanViirok) [inline esteid_certificates/constants.py]: issue: Now this is odd. I'm guessing they should be the other way around? Currently the TEST and DEMO certs are in the `LIVE_CERTS`.

File: autoupdate/update.py [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+            (pathlib.Path(directory_path) / certificate).unlink()
+
+    def update_constants(self, constants_path: str):
+        logging.info("Updating constants file %s", constants_path)
+
+        test_certs = repr(
+            {
+                cert.title: cert.filename
+                for cert in self.certificates
+                if cert.certificate_type != CertificateType.TEST
```
Comment (JohanViirok) [inline autoupdate/update.py]: I'm guessing this should be:
```suggestion
                if cert.certificate_type == CertificateType.TEST
```

File: autoupdate/update.py [hunk 6]
```diff
@@ -0,0 +1,420 @@
...
+            }
+        )
+        live_certs = repr(
+            {
+                cert.title: cert.filename
+                for cert in self.certificates
+                if cert.certificate_type != CertificateType.INTERMEDIATE
```
Comment (JohanViirok) [inline autoupdate/update.py]: And this one?
```suggestion
                if cert.certificate_type == CertificateType.INTERMEDIATE
```

File: esteid_certificates/files/EE_Certification_Centre_Root_CA.pem [hunk 7]
```diff
@@ -1,24 +1,24 @@
...
+VR0TAQH/BAUwAwEB/zAOBgNVHQ8BAf8EBAMCAQYwHQYDVR0OBBYEFBLyWj7qVhy/
+zQas8fElyalL1BSZMEUGA1UdJQQ+MDwGCCsGAQUFBwMCBggrBgEFBQcDAQYIKwYB
+BQUHAwMGCCsGAQUFBwMEBggrBgEFBQcDCAYIKwYBBQUHAwkwDQYJKoZIhvcNAQEF
+BQADggEBAHv25MANqhlHt01Xo/6tu7Fq1Q+e2+RjxY6hUFaTlrg4wCQiZrxTFGGV
+v9DHKpY5P30osxBAIWrEr7BSdxjhlthWXePdNl4dp1BUoMUq5KqMlIpPnTX/dqQG
+E5Gion0ARD9V04I8GtVbvFZMIi5GQ4okQC3zErg7cBqklrkar4dBGmoYDQZPxz5u
+uSlNDUmJEYcyW+ZLBMjkXOZ0c5RdFpgTlf7727FE5TpwrDdr5rMzcijJs1eg9gIW
+iAYLtqZLICjU3j2LrTcFU3T+bsy8QxdxXvnFzBqpYe73dgzzcvRyrc9yAjYHR8/v
+GVCJYMzpJJUPwssd8m92kMfMdcGWxZ0=
+-----END CERTIFICATE-----
```
Comment (JohanViirok) [inline esteid_certificates/files/EE_Certification_Centre_Root_CA.pem:24]: Github, go home, you're drunk. There is no difference here.

**Review by JohanViirok — CHANGES_REQUESTED** [review]

File: esteid_certificates/files/EE_Certification_Centre_Root_CA.pem (hunk 7 above)
Comment (iharthi) [reply esteid_certificates/files/EE_Certification_Centre_Root_CA.pem:24]: Yeah, confusing. Pycharm does better job explaining this diff. 

Line endings are different (windows vs linux).

File: autoupdate/update.py (hunk 5 above)
Comment (iharthi) [reply autoupdate/update.py]: Yup, that is what is causing the oddities with LIVE_CERTS containing test certificates and vice-versa.

**Review by Jyrno42 — APPROVED** [review]

**Review by JohanViirok — APPROVED** [review]

**Pull request merged**